decimal/decimal_test.go

```diff
@@ -0,0 +1,602 @@
+package decimal_test
+
+import (
+	"encoding/hex"
+	"fmt"
+	"log"
+	"os"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/shopspring/decimal"
+	. "github.com/tarantool/go-tarantool"
+	. "github.com/tarantool/go-tarantool/decimal"
+	"github.com/tarantool/go-tarantool/test_helpers"
+	"gopkg.in/vmihailenco/msgpack.v2"
+)
+
+// There is no way to skip tests in testing.M,
+// so we use this variable to pass info
+// to each testing.T that it should skip.
+var isDecimalSupported = false
+
+var server = "127.0.0.1:3013"
+var opts = Opts{
+	Timeout: 500 * time.Millisecond,
+	User:    "test",
+	Pass:    "test",
+}
+
+var space = "testDecimal"
+var index = "primary"
+
+type TupleDecimal struct {
+	number decimal.Decimal
+}
+
+func (t *TupleDecimal) EncodeMsgpack(e *msgpack.Encoder) error {
+	if err := e.EncodeSliceLen(1); err != nil {
+		return err
+	}
+	return e.EncodeValue(reflect.ValueOf(t.number))
+}
+
+func (t *TupleDecimal) DecodeMsgpack(d *msgpack.Decoder) error {
+	var err error
+	var l int
+	if l, err = d.DecodeSliceLen(); err != nil {
+		return err
+	}
+	if l != 1 {
+		return fmt.Errorf("Array length doesn't match: %d", l)
+	}
+
+	res, err := d.DecodeInterface()
+	if err != nil {
+		return err
+	}
+	t.number = res.(decimal.Decimal)
+
+	return nil
+}
+
+var benchmarkSamples = []struct {
+	numString string
+	mpBuf     string // MessagePack buffer
+	fixExt    bool   // Whether it is FixExt or Ext MessagePack type.
+}{
+	// MessagePack, fixext 1, 0xd4 (TODO: no sample)
+	{"", "", true},
+	// MessagePack, fixext 2, 0xd5
+	{"0.3", "d501013c", true},
+	// MessagePack, fixext 4, 0xd6
+	{"-18.34", "d6010201834d", true},
+	// MessagePack, fixext 8, 0xd7
+	{"-108.123456789", "d701090108123456789d", true},
+	// MessagePack, fixext 16, 0xd8 (TODO: no sample)
+	{"", "", true},
```

Comment:
If test cases do nothing, shouldn't we exclude them from sample list at all (e.g. comment `{"", "", bool}` rows)?

(I see that there is a skip in test if sample is `""`, but not adding it at all seems more appropriate to me.)

Reply:
No. It is better to leave them for clarity. It is better to know what tests are not cover than not.